experimental/WreathMacdonaldpols/src/WreathMacdonaldpols.jl

```diff
@@ -0,0 +1,241 @@
+all multipartition of size n and length r in the standard Schur basis indexed by the
+multipartitions of the same size and length as lbb. Each row of this matrix is a wreath
+Macdonald polynomial.
+"""
+function wreath_macdonald_polynomials(n::Int,
+                                      r::Int,
+                                      wperm::PermGroupElem,
+                                      coroot::Vector{Int};
+                                      parent::AbstractAlgebra.Generic.MPolyRing{QQAbFieldElem{AbsSimpleNumFieldElem}})
+
+  Q = fraction_field(parent)
+  q,t = gens(parent)
+
+  mps=collect(multipartitions(n,r))
+  l=length(mps)
+
+  c_L=C_L(r,n,Q,t,mps)
+  c_L_q=map_entries(f->numerator(f)(0,q)//denominator(f)(0,q),c_L)
+  c_L_tinv=map_entries(f->numerator(f)(0,1//t)//denominator(f)(0,1//t),c_L)
+
+  rows=[]
+  for lbb in mps
+    smallers=smaller_ord(lbb, wperm, coroot,mps) #tinv
+    biggers=bigger_ord(lbb, wperm, coroot,mps) #q
+    smaller_indices=sort!(indexin(smallers,mps))
+    bigger_indices=sort!(indexin(biggers,mps))
+    sub_smaller_tinv=vcat(map(i->c_L_tinv[i:i,:],smaller_indices)...)
+    sub_bigger_q=vcat(map(i->c_L_q[i:i,:],bigger_indices)...)
+    M=vcat(sub_smaller_tinv,sub_bigger_q)
+    B=kernel(M, side=:left)
+    rows=vcat(rows,B[1,1:length(smaller_indices)]*sub_smaller_tinv)
+  end
+  c_L_qt = matrix(Q,l,l,rows)
+
+  triv=[partition([n])]
+  for i in 2:r
+    push!(triv,partition([]))
+  end
+  triv=multipartition(triv)
+  index_triv=findfirst(x-> x==triv,mps)
+  d=[1//c_L_qt[i,index_triv] for i in 1:l]
+  diag=diagonal_matrix(Q,d)
+  c_L_qt_H=diag*c_L_qt
+  return c_L_qt_H
+end
+
+@doc raw"""
+
+    wreath_macdonald_polynomial(lbb::Multipartition,
+                                wperm::PermGroupElem,
+                                coroot::Vector{Int};
+                                parent::AbstractAlgebra.Generic.MPolyRing{QQAbFieldElem{AbsSimpleNumFieldElem}})
+
+Given a multipartition lbb and an element of the affine Weyl group of type A (seen as
+the semi-direct product of the Symmetric group with the coroot lattice), this function
+returns the coefficients of the wreath Macdonald polynomial associated with lbb and
+the affine Weyl group element in the standard Schur basis indexed by the multipartitions
+ of the same size and length as lbb.
+"""
+function wreath_macdonald_polynomial(lbb::Multipartition,
+                                     wperm::PermGroupElem,
+                                     coroot::Vector{Int};
+                                     parent::AbstractAlgebra.Generic.MPolyRing{QQAbFieldElem{AbsSimpleNumFieldElem}})
+
+  Q = fraction_field(parent)
+  q,t = gens(parent)
+
+  r=length(lbb)
+  n=sum(lbb)
+  mps=collect(multipartitions(n,r))
+  l=length(mps)
+
+  c_L=C_L(r,n,Q,t,mps)
+  c_L_q=map_entries(f->numerator(f)(0,q)//denominator(f)(0,q),c_L)
+  c_L_tinv=map_entries(f->numerator(f)(0,1//t)//denominator(f)(0,1//t),c_L)
```

Comment:
We should enhance AbstractAlgebra / Nemo to allow replacing this code by this:
```suggestion
  c_L_q=map_entries(f->f(0,q),c_L)
  c_L_tinv=map_entries(f->f(0,1//t),c_L)
```
Alas, right now this doesn't work. (Perhaps it might work with `evaluate`, but let's not delay this PR further over this)

Reply:
See also https://github.com/Nemocas/AbstractAlgebra.jl/pull/2090